aten/src/THS/generic/THSTensor.cpp

```diff
@@ -146,8 +146,14 @@ THSTensor *THSTensor_(newWithTensorAndSize)(THLongTensor *indices, THTensor *val
   THSTensor *self = (THSTensor *)THAlloc(sizeof(THSTensor));
   THSTensor_(rawInit)(self);
 
-  nDimI = THLongTensor_size(indices, 0);
-  nDimV = THTensor_(nDimension)(values) - 1;
+  // TODO: we may need to special case when only one of these are empty.
+  if (THLongTensor_nDimension(indices) == 0 && THTensor_(nDimension)(values) == 0) {
+    nDimI = 0;
+    nDimV = THLongStorage_size(sizes);
```

Comment:
I think there should be a check for if sizes is NULL here. Let's say that someone omits the sizes field in the tensor constructor:
`torch.sparse.FloatTensor(torch.LongTensor(), torch.FloatTensor())`. Then `sizes` would be NULL and this would be UB.

Reply:
That sounds right, although I think the effect will be the same -- we can't actually figure out nDimI/nDimV in that case, so either the old or new code path will break.